README.md

```diff
@@ -103,15 +108,102 @@ Rlog is configured via the following environment variables:
+                    config file has changed. By default, this is set to 15
+                    seconds. This means that within 15 seconds a changed
+                    logging configuration will take effect. Note that this
+                    check is only performed when a log message is actually
+                    written. If the program does nothing or doesn't log
+                    messages, the config file won't be read. If there is no
+                    config file or it has been removed then the configuration
+                    from the environment variables is used. Set this value to
+                    0 order to switch off the regular log file checking. Be
+                    careful when setting it to 0 via the config file: rlog will
+                    not check the config file again and therefore, this setting
+                    cannot be undone.
 
 Please note! If these environment variables have incorrect or misspelled
 values then they will be silently ignored and a default value will be used.
 
 
+## Using the config file
+
+A config file for rlog is entirely optional, since rlog works just fine even
+without it. However, it does provide you with a very neat feature: You can
+change the logging configuration of a running program without having to restart
+it!
+
+When rlog is imported it starts out with the defaults described above. It then
+takes an initial configuration from environment variables, which may override
+the default values. Next, it looks for the rlog config file. If it cannot find
+the config file it will quietly continue without error. If the config file is
+found then the configuration from environment variables is combined with the
+configuraton from the config file. More about how this combination works, and
+what takes precedence, in a moment.
+
+### Updating the logging config of a running program
+
+Every time you log a message and at least RLOG_CONF_CHECK_INTERVAL seconds have
+elapsed since the last reading of the config file, rlog will automatically
+re-read the content of the conf file and re-apply the configuration it finds
+there over the initial configuration, which was based on the environment
+variables.
+
+You can always just delete the config file to go back to the configuration
+based on environment variables.
+
+### Logfile location
+
+The absolute path for the config file can be set via the RLOG_CONF_FILE
+environment variable. Absent that, rlog looks for a config file in
+"/etc/rlog/your-executable-name.conf". This means that you can easily provide
+different logging configurations for each of your processes.
+
+### Logfile format
+
+The format of the logfile is simple. Each setting is referred to by the same
+name as the environment variable. So, your config file may look like this:
+
+    RLOG_LOG_LEVEL=WARN
+    RLOG_LOG_STREAM=stdout
+    RLOG_TIME_FORMAT=UnixDate
+    RLOG_LOG_FILE=/var/log/myapp.log
+
```

Comment:
No comments allowed, but empty lines and leading or trailing spaces in lines are ignored. Lines cannot be split, so the value for a setting needs to fit into a single line.

Line is split by the first '='. So, if you have more '=' (or spaces) after that, it will all come through as a single token. Therefore, filenames with spaces or '=' should be fine.

Only issue: Spaces of each token are also stripped. For example:

    '      FOO BAR_BLAH    =    ONE TWO=  THREE     '

That line (start and end marked by ') would be parsed into two tokens:

    'FOO BAR_BLAH'
    'ONE TWO= THREE'

I think this behaviour should suffice, no?